src/StandaloneHydratorPluginManager.php

```diff
@@ -0,0 +1,232 @@
+ * order to create the instance (generally only needed if you want to configure
+ * an existing hydrator type or need to use dependencies from the creation
+ * context).
+ *
+ * You can create an alias for a fully qualified class name or factory-based
+ * service using addAlias(). Aliases MUST resolve to either a factory-based
+ * service or fully qualified class name.
+ *
+ * Instances returned by the plugin manager are never cached; new instances are
+ * always returned. This is because the various instances provided in the
+ * package are mutable, and strategies, naming strategies, and filters may need
+ * to differ between instances.
+ */
+class StandaloneHydratorPluginManager implements
+    ContainerInterface,
+    HydratorPluginManagerInterface
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $creationContext;
+
+    /**
+     * @var array<string, string>
+     */
+    private $aliases = [
+        'arrayserializable'  => ArraySerializable::class,
+        'classmethods'       => ClassMethods::class,
+        'delegatinghydrator' => DelegatingHydrator::class,
+        'objectproperty'     => ObjectProperty::class,
+        'reflection'         => Reflection::class,
+    ];
+
+    /**
+     * @var array<string, string|callable>
+     */
+    private $factories = [
+        DelegatingHydrator::class => DelegatingHydratorFactory::class,
+    ];
+
+    /**
+     * @var callable
+     */
+    private $invokableFactory;
+
+    /**
+     * @param null|ContainerInterface $creationContext Parent container to use
+     *     for retrieving dependencies, if needed.
+     */
+    public function __construct(?ContainerInterface $creationContext = null)
+    {
+        $this->creationContext = $creationContext ?: $this;
+        $this->invokableFactory = function (ContainerInterface $container, string $class) {
+            return new $class();
+        };
+        foreach ($this->aliases as $alias => $class) {
+            $this->factories[$class] = $this->factories[$class] ?? $this->invokableFactory;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function get($id)
+    {
+        $class = $this->resolveName($id);
+        if (! $class) {
+            throw Exception\MissingHydratorServiceException::forService($id);
+        }
+
+        $instance = $this->createViaFactory($class);
+
+        if (! $instance instanceof HydratorInterface) {
+            throw Exception\InvalidHydratorServiceException::forService($id);
+        }
+
+        return $instance;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function has($id)
+    {
+        return null !== $this->resolveName($id);
+    }
+
+    /**
+     * @throws Exception\InvalidArgumentException if unable to resolve the
+     *     alias to a previously resolved alias, a service mapped to a
+     *     factory, or a fully qualified class name.
+     */
+    public function addAlias(string $alias, string $target) : void
```

Comment:
Do we need this sort of complexity in the base plugin manager? Assuming we only expose the PSR-11 API when falling back to this container, we can probably leave out most of it, no?

Reply:
I think having something that doesn't allow adding services makes it pretty much useless, as the point is to allow pulling either specific types, or specifically configured instances, and the latter two are likely to be user-generated. If we don't have _some_ capabilities for adding to the manager, users are then forced to create their own implementation as soon as they need to use a custom hydrator, or want one of the shipped hydrators with specific customisations (strategies, filters, etc.).

We _could_ limit it to only factories, but that can potentially lead to a lot of duplicated service definitions. I felt limiting to aliases, invokables (via aliases), and factories was a reasonable compromise in this situation.

Reply:
> I think having something that doesn't allow adding services makes it pretty much useless

It's a fallback DIC for setups where a configurable DIC is not wanted. This is the scenario where users just want the hydrators internals to work without worrying too much about them.

I'd even just hardcode the map of services in an inlined `get` and `has` in a minimal PSR-11 container implementation here...

Reply:
Fair enough; if they want to do more, they can install zend-servicemanager, wire them into the container they use normally, or create their own container.

I'll update the PR this morning to do that, and document the approaches (including an example of how to write a custom container).